airflow/api_fastapi/core_api/serializers/tasks.py

```diff
@@ -0,0 +1,159 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import inspect
+from collections import abc
+from datetime import datetime, timedelta
+
+from pydantic import AliasGenerator, BaseModel, ConfigDict, computed_field, field_validator
+
+from airflow.models.mappedoperator import MappedOperator
+from airflow.serialization.serialized_objects import SerializedBaseOperator, encode_priority_weight_strategy
+from airflow.task.priority_strategy import PriorityWeightStrategy
+
+
+class TimeDeltaResponse(BaseModel):
+    """Time delta serializer for responses."""
+
+    object_type: str = "TimeDelta"
+    days: int
+    seconds: int
+    microseconds: int
+
+    model_config = ConfigDict(
+        alias_generator=AliasGenerator(
+            serialization_alias=lambda field_name: {
+                "object_type": "__type",
+            }.get(field_name, field_name),
+        )
+    )
+
+
+class TaskResponse(BaseModel):
+    """Task serializer for responses."""
+
+    task_id: str | None
+    task_display_name: str | None
+    owner: str | None
+    start_date: datetime | None
+    end_date: datetime | None
+    trigger_rule: str | None
+    depends_on_past: bool
+    wait_for_downstream: bool
+    retries: float | None
+    queue: str | None
+    pool: str | None
+    pool_slots: float | None
+    execution_timeout: TimeDeltaResponse | None
+    retry_delay: TimeDeltaResponse | None
+    retry_exponential_backoff: bool
+    priority_weight: float | None
+    weight_rule: str | None
+    ui_color: str | None
+    ui_fgcolor: str | None
+    template_fields: list[str] | None
+    downstream_task_ids: list[str] | None
+    doc_md: str | None
+    operator_name: str | None
+    params: abc.MutableMapping | None
+    class_ref: dict | None
+    is_mapped: bool | None
+
+    @field_validator("weight_rule", mode="before")
+    @classmethod
+    def validate_weight_rule(cls, wr: str | PriorityWeightStrategy | None) -> str | None:
+        """Validate the weight_rule property."""
+        if wr is None:
+            return None
+        if isinstance(wr, str):
+            return wr
+        return encode_priority_weight_strategy(wr)
+
+    @field_validator("execution_timeout", mode="before")
+    @classmethod
+    def validate_execution_timeout(cls, et: timedelta | None) -> TimeDeltaResponse | None:
+        """Validate the execution_timeout property."""
+        if et is None:
+            return None
+        return TimeDeltaResponse(
+            days=et.days,
+            seconds=et.seconds,
+            microseconds=et.microseconds,
+        )
```

Comment:
This can be factorized with `Annotated[...., BeforeValidator]` notation. (same for retry_delay).

1 common function, used for 3 different fields.

Reply:
Done, moved the annotation to `types.py`, using it for both `timedelta` fields.